src/ReactLeafletMap/index.js

```diff
@@ -87,20 +87,29 @@ class ReactLeafletMap extends React.PureComponent {
     }
 
     onViewportChange(viewport) {
-        const center = {
-            lat: viewport.center[0],
-            lon: viewport.center[1]
-        };
-
-        const boxRange = viewUtils.getBoxRangeFromZoomLevel(viewport.zoom);
+        const change = {};
+        // viewport.center could be undefined
+        if(viewport && viewport.hasOwnProperty('center') && viewport.center) {
+            const center = {
+                lat: viewport.center[0],
+                lon: viewport.center[1]
+            };
+            change['center'] = center;
+        }
+        
+        // viewport.zoom could be undefined
+        if(viewport && viewport.hasOwnProperty('zoom') && Number.isFinite(viewport.zoom)) {
```

Comment:
```suggestion
        if(viewport && viewport.zoom) {
```

Reply:
As in comment viewport.zoom can be undefined